internal/engine/local/controller.go

```diff
@@ -117,14 +117,12 @@ func (c *Controller) stop(name model.ManifestName) {
 	proc := c.procs[name]
 	// change the process's current number so that any further events received by the existing process will be considered out of date
 	proc.incrProcNum()
-	if proc.probeWorker != nil {
-		proc.probeWorker.Stop()
```

Comment:
Ah meant to throw a comment on this - I eliminated the `Stop()` method on the worker - it was buggy & redundant. `Run()` takes a context so that's (already) used to stop the worker via cancelation propagation.